vllm/entrypoints/openai/tool_parsers/hermes_tool_parser.py

```diff
@@ -44,7 +44,8 @@ def __init__(self, tokenizer: AnyTokenizer):
         self.tool_call_end_token: str = "</tool_call>"
 
         self.tool_call_regex = re.compile(
-            r"<tool_call>(.*?)</tool_call>|<tool_call>(.*)", re.DOTALL)
+            r"<tool_call>\s*(.*?)\s*(?:<tool_call>\s*|</tool_call>\s*|$)",
```

Comment:
- `(?:<tool_call>\s*)+` matches `<tool_call>` with or without appended newlines one or more times in a non-capturing group
- `(?P<tool_call>.*)` captures anything and puts it in a named capture group `tool_call`
- `(?:\s</tool_call>)?` matches `</tool_call>` with or without preceding newlines one or more times in a non-capturing group

```suggestion
            r"(?:<tool_call>\s*)+(?P<tool_call>.*)(?:\s</tool_call>)?",
```